scripts/west_commands/build.py

```diff
@@ -41,7 +42,15 @@
 
 west build [...] -- -DOVERLAY_CONFIG=some.conf
 
-(Doing this forces a CMake run.)'''
+(Doing this forces a CMake run.)
+
+positional arguments:
```

Comment:
Having this here is a bit hacky, but the actual output is identical. I went for the KIS approach, I doubt very much argparse is going to change its layout

Reply:
Yeah, that seems reasonable and we can always adapt in later Python versions if it does.

One thing to note is that in west's main.py we go to great lengths to format help properly for terminal widths that are smaller than 75 characters. See the `shutil.get_terminal_size()` call in `WestArgumentParser.format_help()` and the way it is used in `format_thing_and_help()` in the same class for details.

Since this output is not so carefully wrapped, it will look different than the actual options in small terminals.

Do you think it would be easy to fix that here? If so I think it's worth the effort.

Reply:
I looked at this, but here's the issue: The options and arguments are formatted by `ArgParse` and the way it is it actually matches the rest of them, which are actually added automagically by it:

```
positional arguments:
  source_dir            Explicitly set the source directory. If not given and
                        rebuilding an existing Zephyr build directory, this is
                        taken from the CMake cache. Otherwise, the current
                        directory is assumed.
  cmake_opt             Extra options to pass to CMake; implies -c

optional arguments:
  -h, --help            show this help message and exit
  -b BOARD, --board BOARD
                        Board to build for (must be given for the first build,
                        can be omitted later)
  -d BUILD_DIR, --build-dir BUILD_DIR
                        Explicitly sets the build directory. If not given and
                        the current directory is a Zephyr build directory, it
                        will be used; otherwise, "build" is assumed.The
                        directory is created if it doesn't exist.
  -t TARGET, --target TARGET
                        Override the build system target (e.g. 'clean',
                        'pristine', etc.)
  -c, --cmake           Force CMake to run
  -f, --force           Ignore any errors and try to proceed
```

If I now force a smaller terminal size, the output wraps in exactly the same way for the two we forcefully add help for (`source_dir` and `cmake_opt`) and for the rest (added by `ArgParse` itself):

![image](https://user-images.githubusercontent.com/12450381/53816057-2e250100-3f63-11e9-8348-7194d2e818b8.png)